### PR TITLE
KAFKA-9143: Log task reconfiguration error only when it happened

### PR DESCRIPTION
**Pull request by ivanyu**
This commit makes `DistributedHerder` log that some error has happened during task reconfiguration only when it actually has happened.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by rhauch — APPROVED** [review]
Thanks for identifying the problem and issuing a quick fix, @ivanyu. LGTM!

**Comment by ivanyu** [comment]
It needs another approval. I'd like to tag @ewencp as the author of the original piece.

**Comment by rhauch** [comment]
@ivanyu this PR does not need another approval, since I am a committer. However, we're currently in AK 2.4 code freeze, so I was holding off merging since we want to backport this to the `2.4` branch (and further back).

**Review by andrewchoi5 — APPROVED** [review]

**Comment by ivanyu** [comment]
@rhauch any chance merging this now?

**Pull request merged**
